### PR TITLE
Add internal OpenCensus metric translation library

### PR DESCRIPTION
**Pull request by dashpole**
Part of https://github.com/open-telemetry/opentelemetry-go/issues/2808.

This doesn't fully implement the metric bridge, but contains the conversion function for translating OpenCensus metrics to OpenTelemetry.

### CODE REVIEW

**Comment by dashpole** [comment]
On the 1.17 compatibility test, i'm getting:
```
go test -timeout 60s -short ./bridge/opencensus/opencensusmetric/...
go: warning: "./..." matched no packages
no Go files in /home/runner/work/opentelemetry-go/opentelemetry-go/bridge/opencensus/opencensusmetric
```

Seems like it doesn't like a 1.18-only package. Any advice?

**Comment by MrAlias** [comment]
> On the 1.17 compatibility test, i'm getting:
> 
> ```
> go test -timeout 60s -short ./bridge/opencensus/opencensusmetric/...
> go: warning: "./..." matched no packages
> no Go files in /home/runner/work/opentelemetry-go/opentelemetry-go/bridge/opencensus/opencensusmetric
> ```
> 
> Seems like it doesn't like a 1.18-only package. Any advice?

Add a `doc.go` for the package documentation and leave out the build constraint for 1.18. That way the tooling will still find a go file that it can "build".

**Review by hanyuancheung — APPROVED** [review]
Generally lgtm

File: bridge/opencensus/opencensusmetric/internal/metric.go [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+// +build go1.18
+
+package internal // import "go.opentelemetry.io/otel/bridge/opencensus/opencensusmetric/internal"
+
+import (
+	"errors"
+	"fmt"
+
+	ocmetricdata "go.opencensus.io/metric/metricdata"
+
```
[the remark this answers is not in the document]
Comment (dashpole) [reply bridge/opencensus/opencensusmetric/internal/metric.go:25]: Generally, we seem to separate imports into 3 groups:  standard lib, external dependencies, and dependencies from the repository.  The OpenCensus dependency is in the second category, and I believe belongs in its own group.

**Review by MrAlias — APPROVED** [review]

**Review by MadVikingGod — APPROVED** [review]

**Pull request merged**